src/xGadgetron/cGadgetron/gadgetron_data_containers.cpp

```diff
@@ -1140,6 +1143,98 @@ GadgetronImagesVector::set_real_data(const float* data)
 		*iter = *data;
 }
 
+void
+
+bool is_unit_vector(const float *vec)
+{
+    return sqrt(vec[0]*vec[0] + vec[1]*vec[1] + vec[2]*vec[2]) - 1.F < 1.e-7F ? true : false;
```

Comment:
```suggestion
    return std::abs(vec[0]*vec[0] + vec[1]*vec[1] + vec[2]*vec[2] - 1.F) < 1.e-4F;
```
no reason to do the `sqrt`. 1E-7 is really quite hard to achieve with float calculations. need an `abs`